// graphql-toolkit/examples/resolvers-first-basic/dist/post.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const graphql_toolkit_1 = require("graphql-toolkit");
const author_1 = require("./author");
const authors_collection_1 = require("./authors.collection");
let Post = class Post {
    constructor({ id, title, content, authorId }) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.authorId = authorId;
    }
    author() {
        return authors_collection_1.AUTHORS.find(({ id }) => id === this.authorId);
    }
};
tslib_1.__decorate([
    graphql_toolkit_1.InputFieldProperty(),
    graphql_toolkit_1.FieldProperty(),
    tslib_1.__metadata("design:type", Number)
], Post.prototype, "id", void 0);
tslib_1.__decorate([
    graphql_toolkit_1.InputFieldProperty(),
    graphql_toolkit_1.FieldProperty(),
    tslib_1.__metadata("design:type", String)
], Post.prototype, "title", void 0);
tslib_1.__decorate([
    graphql_toolkit_1.InputFieldProperty(),
    graphql_toolkit_1.FieldProperty(),
    tslib_1.__metadata("design:type", String)
], Post.prototype, "content", void 0);
tslib_1.__decorate([
    graphql_toolkit_1.Field(() => author_1.Author),
    tslib_1.__metadata("design:type", Function),
    tslib_1.__metadata("design:paramtypes", []),
    tslib_1.__metadata("design:returntype", author_1.Author)
], Post.prototype, "author", null);
tslib_1.__decorate([
    graphql_toolkit_1.InputFieldProperty(),
    tslib_1.__metadata("design:type", Number)
], Post.prototype, "authorId", void 0);
Post = tslib_1.__decorate([
    graphql_toolkit_1.ObjectType(),
    graphql_toolkit_1.InputObjectType({ name: 'PostInput' }),
    tslib_1.__metadata("design:paramtypes", [Object])
], Post);
exports.Post = Post;
